lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -306,33 +306,12 @@ UsdTimeCode getTime(const Ufe::Path& path)
         return UsdTimeCode::Default();
     }
 
-    // Get the time from the proxy shape.  This will be the tail component of
-    // the first path segment.
-    auto proxyShapePath = Ufe::Path(path.getSegments()[0]);
-
-    // Keep a single-element path to MObject cache, as all USD prims in a stage
-    // share the same proxy shape.
-    static std::pair<Ufe::Path, MObjectHandle> cache;
-
-    MObject proxyShapeObj;
-
-    if (cache.first == proxyShapePath && cache.second.isValid()) {
-        proxyShapeObj = cache.second.object();
-    } else {
-        // Not found in the cache, or no longer valid.  Get the proxy shape
-        // MObject from its path, and put it in the cache.  Pop the head of the
-        // UFE path to get rid of "|world", which is implicit in Maya.
-        auto proxyShapeDagPath = UsdMayaUtil::nameToDagPath(proxyShapePath.popHead().string());
-        TF_VERIFY(proxyShapeDagPath.isValid());
-        proxyShapeObj = proxyShapeDagPath.node();
-        cache = std::pair<Ufe::Path, MObjectHandle>(proxyShapePath, MObjectHandle(proxyShapeObj));
+    // Proxy shape node should not be null.
+    auto proxyShape = g_StageMap.proxyShapeNode(path);
```

Comment:
Removed homebrew, single node caching in favor of using the global proxy shape cache.